neuraxle/pipeline.py

```diff
@@ -62,79 +35,138 @@ def __hash__(self):
+
+        return new_self
+
+    def fit_transform_steps(self, data_inputs, expected_outputs):
+        """
+        After loading the last checkpoint, fit transform each pipeline steps
+        :param data_inputs: the data input to fit on
+        :param expected_outputs: the expected data output to fit on
+        :return: the pipeline itself
+        """
+        steps_left_to_do, data_inputs = self.read_checkpoint(data_inputs)
 
-    def transform(self, data_inputs):
-        return self.pipeline_runner.transform(data_inputs)
+        new_steps_as_tuple: NamedTupleList = []
+        for step_name, step in steps_left_to_do:
+            step, data_inputs = step.fit_transform(data_inputs, expected_outputs)
+            new_steps_as_tuple.append((step_name, step))
 
-    def fit_transform(self, data_inputs, expected_outputs=None) -> ('Pipeline', Any):
-        self.pipeline_runner.set_steps(self.steps_as_tuple)
+        self.steps_as_tuple = self.steps_as_tuple[:len(self.steps_as_tuple) - len(steps_left_to_do)] + \
+                              new_steps_as_tuple
 
-        self.pipeline_runner, out = self.pipeline_runner.fit_transform(data_inputs, expected_outputs)
-        return self, out
+        return self, data_inputs
 
-    def inverse_transform(self, processed_outputs):
-        if self.transform != self.inverse_transform:
-            raise BrokenPipeError("Don't call inverse_transform on a pipeline before having mutated it inversely or "
-                                  "before having called the `.reverse()` or `reversed(.)` on it.")
+    def transform(self, data_inputs):
+        """
+        After loading the last checkpoint, transform each pipeline steps
+        :param data_inputs: the data input to fit on
+        :return: transformed data inputs
+        """
+        steps_left_to_do, data_inputs = self.read_checkpoint(data_inputs)
+        for step_name, step in steps_left_to_do:
+            data_inputs = step.transform(data_inputs)
+
+        return data_inputs
+
+    def inverse_transform_processed_outputs(self, processed_outputs) -> Any:
+        """
+        After transforming all data inputs, and obtaining a prediction, we can inverse transform the processed outputs
+        :param processed_outputs: the forward transformed data input
+        :return: backward transformed processed outputs
+        """
+        for step_name, step in list(reversed(self.steps_as_tuple)):
+            processed_outputs = step.transform(processed_outputs)
+        return processed_outputs
 
-        reversed_steps_as_tuple = list(reversed(self.steps_as_tuple))
-        return self.pipeline_runner.set_steps(reversed_steps_as_tuple).transform(processed_outputs)
+    def read_checkpoint(self, data_inputs) -> Tuple[list, Any]:
+        new_data_inputs = data_inputs
+        new_starting_step_index = self.find_starting_step_index(data_inputs)
+
+        step = self.steps_as_tuple[new_starting_step_index]
```

Comment:
@guillaume-chevalier not sure what you meant here by truncated self ? I think that what I did is fine, but can you show me the syntax to make sure ?

Reply:
To be fixed later I guess.

Reply:
step = self[string_name_of_step]